HRHN/ViewModel/TodayViewModel.swift

```diff
@@ -0,0 +1,26 @@
+//
+//  TodayViewModel.swift
+//  HRHN
+//
+//  Created by Chanhee Jeong on 2022/12/25.
+//
+
+import Foundation
+
+final class TodayViewModel {
+    
+    var todayChallenge: Observable<String?> = Observable(nil)
+    
+    private var coreDataManager = CoreDataManager.shared
+    
+    init(){}
+    
+    func fetchTodayChallenge() {
+        let challenges = self.coreDataManager.getChallengeOf(Date())
+        if challenges.count > 0 {
+            self.todayChallenge = Observable(challenges[0].content)
+        } else {
+            self.todayChallenge = Observable(nil)
+        }
+    }
```

Comment:
어차피 '하루하나'니까 CoreDataManager의 getChallengeOf 함수에서 여러 챌린지가 반환될 일이 없잖아요?
그니까 getChallengeOf가 `Challenge?` 타입을 반환하게 만들어 결과가 있을 땐 배열 말고 Challenge 타입을, 결과가 없을 땐 `nil`을 반환하면 이 부분이 간단해지는데 어떤가요?
```suggestion
    func fetchTodayChallenge() {
        let challenge = self.coreDataManager.getChallengeOf(Date())
        self.todayChallenge = Observable(challenge?.content) // 없으면 알아서 nil이 들어감 아마도?
    }
```

여기 if문 보니까 'getChallengeOf는 어차피 하나만 나올텐데 count > 0이 왜 있지?'라는 생각이 들어서 한 번 던지고 가봅니다!

Reply:
놀랍게도 현재는 getChallengesOf 하면 배열로 나옵니다..!?! 😅 
Challenge 하나만 내보내려하면 그냥 build fail 떠서 일단 저렇게 했는데,, 나중에 왜그런지 뜯어고치면서 바꿀게요ㅠ